src/digest.c

```diff
@@ -103,32 +103,52 @@ SEXP is_little_endian(void) {
 }
 
 /* for any algorithm that has output as unsigned char*, can use _store_from_char_ptr */
-#define _store_from_char_ptr(WHAT) if (leaveRaw) {                          \
-    memcpy(output, (WHAT), output_length);                     \
-} else for (int j = 0; j < output_length; j++) {               \
-    snprintf(output + j * 2, 3, "%02x", (WHAT)[j]);            \
-}                                                              \
-
-#define decl_tmp32(WHAT) uint32_t tmp = (WHAT);
-#define decl_tmp64(WHAT) uint64_t tmp = (WHAT);
+void _store_from_char_ptr(
+    const unsigned char *hash, unsigned char *output,
+    const size_t output_length, const int leaveRaw
+) {
+    if (leaveRaw) {
+        memcpy(output, hash, output_length);
+    } else for (int j = 0; j < output_length; j++) {
+// one char is 2 hex digits => converting to (0-9A-F)-chars == writing 2 spots
```

Comment:
Very helpful comment, also appreciated!.  My editor would align the `//` with code, though :wink: